taskchampion/lib/src/server.rs

```diff
@@ -170,6 +170,7 @@ pub unsafe extern "C" fn tc_server_new_sync(
 #[no_mangle]
 pub unsafe extern "C" fn tc_server_new_gcp(
     bucket: TCString,
+    credentialpath: Option<TCString>,
```

Comment:
Since this function is called from C++, it can't use Rust-specific types like `Option`. This will need to be just a plain `TCString`.

Reply:
Ah I forgot to push the new changes, what do you think of this?
```rust
#[no_mangle]
pub unsafe extern "C" fn tc_server_new_gcp( 
    bucket: TCString,
    credential_path_argument: TCString,
    encryption_secret: TCString,
    error_out: *mut TCString,
) -> *mut TCServer {
    wrap(
        || {
            // SAFETY:
            //  - bucket is valid (promised by caller)
            //  - bucket ownership is transferred to this function
            let bucket = unsafe { TCString::val_from_arg(bucket) }.into_string()?;

            // SAFETY:
            //  - credential_path is valid (promised by caller)
            //  - credential_path ownership is transferred to this function
            // TODOAK003: push changes after confirming with @djmitche
            let credential_path:String;

            if credential_path_argument.is_null() {
                credential_path = String::from("");
            } else {
                credential_path = unsafe { TCString::val_from_arg(credential_path_argument) }.into_string()?;
            }

            // SAFETY:  
 
```

Reply:
Close, but `credential_path` should be an `Option<String>`.